### ISSUE TITLE
Any way to stop/skip futher traversal with ESLintUtils.RuleCreator?

### ISSUE THREAD

**Issue by ericbiewener** [typescript-eslint/typescript-eslint#2734]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

Production code:

```tsx
const fn = () => fetch("foo")`
```

Relevant custom rule code:

```tsx
  create(context) {
    return {
      "VariableDeclaration Identifier[name=fetch]"(node: TSESTree.Identifier) {
        if (
          node.parent?.type === AST_NODE_TYPES.Property &&
          node.parent.key === node
        ) {
          // { fetch: ... }
          return;
        }

        if (!doesNonGlobalFetchExist(context.getScope())) {
          context.report({
            messageId,
            node,
          });
        }
      },
      "CallExpression[callee.name=fetch]"(node) {
        if (!doesNonGlobalFetchExist(context.getScope())) {
          context.report({
            messageId,
            node,
          });
        }
      },
    };
  },
```

**Expected Result**

Custom rule reports only once for the given production code

**Actual Result**

It reports twice, because both selectors run.

**Additional Info**

I'm looking for a way to stop traversal, [like there is for @babel/traverse](https://github.com/jamiebuilds/babel-handbook/blob/master/translations/en/plugin-handbook.md#stopping-traversal).

**Versions**

| package                                 | version |
| --------------------------------------- | ------- |
| `@typescript-eslint/experimental-utils` | `2.34.0` |
| `TypeScript`                            | `3.9.7` |
| `node`                                  | `14.11.0` |

**Comment by bradzacher** [comment]
For reference - we don't maintain the ESLint tooling - we just provide some typed utilities on top of it.

Each selector is evaluated independently and the selector function returns `void` - there's no way to short-circuit traversal.
This is because every single lint rule is evaluated simultaneously on the file.

If you want to prevent double reports - you'll have to do this manually in your lint rule.

**Comment by ericbiewener** [comment]
Gotcha, thanks for the clarification!
